src/api/variables/mutate/updateVariable.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  VarFormSchemaType,
+  VarUpdatedSchema,
+  VarUpdatedSchemaType,
+} from "../schema";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+
+import { apiFactory } from "~/api/utils";
+import { useApiKey } from "~/util/store/apiKey";
+import { useNamespace } from "~/util/store/namespace";
+import { useToast } from "~/design/Toast";
+import { varKeys } from "..";
+
+const updateVar = apiFactory({
```

Comment:
This is used for creating and updating variables because the backend API implementation is the same for both cases.